besu/src/main/java/org/hyperledger/besu/controller/TargetingGasLimitCalculator.java

```diff
@@ -56,17 +57,23 @@ public void changeTargetGasLimit(final Long targetGasLimit) {
     this.targetGasLimit = targetGasLimit;
   }
 
+  private long adjustAmount(final long gasLimit) {
+    final long maxAdjustAmount =
+        Math.max(Long.divideUnsigned(gasLimit, GASLIMIT_BOUND_DIVISOR) - 1, 0);
+    return Math.min(ADJUSTMENT_FACTOR, maxAdjustAmount);
+  }
+
```

Comment:
Since we already have logic like this in the `Rule` class, I think we should find a common place for this validation that they both use.

Reply:
Thanks for suggestion
 I think I'll move duplicated logic to a static method, maybe under `GasLimitRangeAndDeltaValidationRule`?

Reply:
Sounds good to me!